### PR TITLE
Print datasets stat

### PR DESCRIPTION
**Pull request by ljj7975**
- utilizing Vocab and WakeWordTokenizer for WakeWordSearcher
- added count_vocab functionality for WakeWordSearcher
- added test cases for classes in tokenizer.py ad WakeWordSearcher
- searching with lower cases

```
AudioDatasetStatistics(num_examples=26634, audio_length_seconds=111928.04543750011, vocab_counts=Counter({'hey': 4157, 'fire': 2420, 'fox': 2124}))
AudioDatasetStatistics(num_examples=2607, audio_length_seconds=10679.505062500015, vocab_counts=Counter({'hey': 564, 'fire': 294, 'fox': 269}))
AudioDatasetStatistics(num_examples=2558, audio_length_seconds=10364.291000000001, vocab_counts=Counter({'hey': 465, 'fire': 298, 'fox': 265}))
```

### CODE REVIEW

**Comment by ljj7975** [comment]
close #57

File: howl/data/tokenize.py [hunk 1]
```diff
@@ -49,7 +48,7 @@ def add_word(self, word: str):
     def max_split(self, tokens: Union[List[str], str]) -> Tuple[Union[List[str], str], Union[List[str], str]]:
         node = self.root
         counter = 0
-        for tok in tokens:
+        for tok in tokens.lower():
```
Comment (edwinzhng) [inline howl/data/tokenize.py:51]: Does `.lower()` work properly if tokens is a list?

File: test/data/tokenize.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
+import unittest
+
+from howl.data.tokenize import Vocab, VocabTrie, WakeWordTokenizer
+
+
+class TestVocab(unittest.TestCase):
```
Comment (edwinzhng) [inline test/data/tokenize.py:6]: Thanks for the tests! Maybe we can add a `.travis.yml` file to run CI for us like in Anserini? (https://github.com/castorini/anserini/blob/master/.travis.yml)

Comment (ljj7975) [reply test/data/tokenize.py:6]: I have a separate ticket for CI  #60

Comment (edwinzhng) [reply test/data/tokenize.py:6]: Oh I see thanks

File: howl/data/tokenize.py (hunk 1 above)
Comment (ljj7975) [reply howl/data/tokenize.py:51]: The notation was incorrect.
I think ralph initially thought about triggering this function from multiple places for different purposes.
However, this is called from only once place right now and takes in string

I have updated the types, if supporting List[str] is necessary in the future, I will add it back

Comment (edwinzhng) [reply howl/data/tokenize.py:51]: Ok great

**Review by edwinzhng — CHANGES_REQUESTED** [review]
Lgtm!

**Review by edwinzhng — APPROVED** [review]

**Comment by ljj7975** [comment]
I found that the strict filtering on the keyword have reduced the number of samples.

I will double-check where the difference is coming from before merging it.

it's likely from dropping samples like `fireflies` cases for vocab `fire`

**Comment by ljj7975** [comment]
as expected, samples that are no longer included for `hey fire fox` have following transcripts
```
the indoor `fire`house is on the stairwell
although camping is unrestricted wood is scarce and camp`fire`s are discouraged
`fox`worth played college football for the university of maryland
```

**Pull request merged**